### ISSUE TITLE
User guide notebook detailing use of Matplotlib to produce plots

### ISSUE THREAD

**Issue by hakonanes** [pyxem/orix#332]
@emichr asked how to include for example a `CrystalMap` plot with scalebar and an inverse pole figure in two separate axes but in the same figure. This isn't shown in the documentation, but is easily done using the Matplotlib registered custom "projections" written in orix:

```python
import matplotlib.pyplot as plt

from orix import plot
from orix.crystal_map import CrystalMap
from orix.quaternion import Orientation, symmetry
from orix.vector import Vector3d


plt.rcParams.update({"font.size": 15, "figure.figsize": (10, 5)})

xmap = CrystalMap.empty((20, 30), step_sizes=(1, 1))
xmap.scan_unit = "nm"

g = Orientation.random(50)
g.symmetry = symmetry.Oh

v = Vector3d.zvector()

fig = plt.figure()
ax0 = fig.add_subplot(121, projection="plot_map")
ax0.plot_map(xmap, xmap.x)
ax0.remove_padding()
ax1 = fig.add_subplot(
    122,
    projection="ipf",
    symmetry=g.symmetry,
    direction=v,
    hemisphere="upper"
)
ax1.set_title(str(v.data.squeeze()), fontweight="bold")
ax1.scatter(g)

fig.savefig("/home/hakon/kode/orix_test/crystal_map_ipf_plot_same_figure.png", bbox_inches="tight", pad_inches=0, dpi=150)
```

![crystal_map_ipf_plot_same_figure](https://user-images.githubusercontent.com/12139781/167644684-2dcb20f2-5784-433f-a201-85a94a959a24.png)

I suggest to add a new user guide topic "Plotting" where "Using orix with Matplotlib" can be explained in more detail, going beyond the convenience methods like `CrystalMap.plot()`, `Vector3d.plot()`, and `Orientation.plot()`, which are covered in other user guides.

**Comment by hakonanes** [comment]
> it seems that adding axes for showing color keys (such as IPFColorKeyTSL) is more complicated

Yes, it is. The relevant function is [here](https://github.com/pyxem/orix/blob/ecfb82dcf9856cf46b05b5dce398ba10c9551cd7/orix/plot/direction_color_keys/direction_color_key_tsl.py#L76): an RGB grid within the Laue symmetry's fundamental sector is created (e.g. within the triangle bounded by [001], [101], and [111] for m-3m), the IPF `matplotlib` axis is set up, and the RGB grid is plotted as an image (`imshow`) into the IPF axis.

Two things that would make this simpler:
* Factor out the RGB grid generation into it's own method to `DirectionColorKey`, so this can be generated without having to plot the color key.
* Get the fundamental sector extent without having get the x/y min/max of the stereographic projection of the vectors within the sector.

As a temporary solution, we can do this manually:

```python
import matplotlib.pyplot as plt
from orix import plot
from orix.crystal_map import CrystalMap
from orix.projections import StereographicProjection
from orix.quaternion import Orientation, symmetry

xmap = CrystalMap.empty((20, 30), step_sizes=(1, 1))
xmap.scan_unit = "nm"

g = Orientation.random(50)
g.symmetry = symmetry.C6h  # 6/m

# Get IPF color key as RGB array
ckey = plot.IPFColorKeyTSL(g.symmetry)
fig = ckey.plot(return_figure=True)
rgb_grid = fig.axes[0].images[0].get_array()
plt.close(fig)

# Get extent of fundamental sector (IPF)
sector = g.symmetry.fundamental_sector
x, y = StereographicProjection().vector2xy(sector.edges)
x_min, x_max = x.min(), x.max()
y_min, y_max = y.min(), y.max()

fig = plt.figure()
ax0 = fig.add_subplot(121, projection="plot_map")
ax0.plot_map(xmap, xmap.x)
ax0.remove_padding()
ax1 = fig.add_subplot(
    122,
    projection="ipf",
    symmetry=g.symmetry,
)
ax1.imshow(rgb_grid, extent=(x_min, x_max, y_min, y_max), zorder=0)

fig.savefig(os.path.join(dir_test, "crystal_map_ipf_ckey_same_figure.png"), **savefig_kwargs)
```

![crystal_map_ipf_ckey_same_figure](https://user-images.githubusercontent.com/12139781/168109991-497424b1-c631-4a06-bb7f-07af7c24a4d0.png)

**Comment by hakonanes** [comment]
@emichr, @harripj has now added a method `InversePoleFigurePlot.plot_ipf_color_key()` which reduces the example in the previous comment to:

```python
import matplotlib.pyplot as plt
from orix import plot
from orix.crystal_map import CrystalMap
from orix.projections import StereographicProjection
from orix.quaternion import symmetry


xmap = CrystalMap.empty((20, 30), step_sizes=(1, 1))
xmap.scan_unit = "nm"

pg = symmetry.C6h  # 6/m

fig = plt.figure()
ax0 = fig.add_subplot(121, projection="plot_map")
ax0.plot_map(xmap, xmap.x)
ax0.remove_padding()
ax1 = fig.add_subplot(122, projection="ipf", symmetry=pg)
ax1.plot_ipf_color_key(show_title=False)
fig.tight_layout()
```

![crystal_map_ipf_ckey_same_figure2](https://user-images.githubusercontent.com/12139781/177264723-c16d824b-5cd5-4143-9644-f52f1539c188.png)

This will be released in the next minor, 0.10. We haven't set a release date yet.
